pkg/reconciler/instances/ory/jwks/jwks.go

```diff
@@ -0,0 +1,88 @@
+package jwks
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/google/uuid"
+	jose "github.com/square/go-jose/v3"
+)
+
+type jwksPatchJSON struct {
+	Op    string             `json:"op"`
+	Path  string             `json:"path"`
+	Value jwksPatchJSONValue `json:"value"`
+}
+type jwksPatchJSONValue struct {
+	Jwks []byte `json:"jwks.json"`
+}
+
+// PreparePatchData generates a JSON Web Key Set with RSA Signature Algorithm and returns the JSON encoded patch for Ory secret.
+func PreparePatchData(alg string, bits int) ([]byte, error) {
```

Comment:
Wouldn't it be simpler to move to an object oriented appraoch instead passing  `alg` + `bits` between all methods?

Like

      type JWKS struct {
          alg string
          bits int
      }

     func NewJWKS(alg string, bits int) *JWKS {
        return &JWKS{alg,bits}
      }

      func (j *JWSK) PreaprePatchData() {
         //take alg + bits from j reference: j.alg. + j.bits 
      }

      func (j *JWSK) generateJwksSecret() {
         //take alg + bits from j reference: j.alg. + j.bits 
      }

Reply:
👍🏻